modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java

```diff
@@ -235,4 +288,209 @@ public static boolean recoverable(Throwable t) {
+     * @param partNum Number of the partition.
+     * @param partId Partition's raft group id..
+     * @param event Assignments switch reduce change event.
+     * @return Completable future that signifies the completion of this operation.
+     */
+    public static CompletableFuture<Void> handleReduceChanged(MetaStorageManager metaStorageMgr, Collection<ClusterNode> baselineNodes,
+            int partitions, int replicas, int partNum, String partId, WatchEvent event) {
+        Entry entry = event.entryEvent().newEntry();
+        byte[] eventData = entry.value();
+
+        List<ClusterNode> assignments = AffinityUtils.calculateAssignments(baselineNodes, partitions, replicas).get(partNum);
+        List<ClusterNode> switchReduce = (List<ClusterNode>) ByteUtils.fromBytes(eventData);
+
+        ByteArray pendingKey = pendingPartAssignmentsKey(partId);
+
+        List<ClusterNode> pendingAssignments = subtract(assignments, switchReduce);
+
+        byte[] pendingByteArray = ByteUtils.toBytes(pendingAssignments);
+        byte[] assignmentsByteArray = ByteUtils.toBytes(assignments);
+
+        if (switchReduce.isEmpty()) {
+            return CompletableFuture.completedFuture(null);
+        }
+
+        ByteArray changeTriggerKey = partChangeTriggerKey(partId);
+        byte[] rev = ByteUtils.longToBytes(event.entryEvent().newEntry().revision());
+
+        // if ((notExists(changeTriggerKey) || value(changeTriggerKey) < revision) && (notExists(pendingKey) && notExists(stableKey)) {
+        //     put(pendingKey, pending)
+        //     put(stableKey, assignments)
+        //     put(changeTriggerKey, revision)
+        // } else if ((notExists(changeTriggerKey) || value(changeTriggerKey) < revision) && (notExists(pendingKey))) {
+        //     put(pendingKey, pending)
+        //     put(changeTriggerKey, revision)
+        // }
+
+        If iif = If.iif(
+                        and(
+                                or(notExists(changeTriggerKey), value(changeTriggerKey).lt(rev)),
+                                and(notExists(pendingKey), (notExists(stablePartAssignmentsKey(partId))))
+                        ),
+                        Operations.ops(
+                                put(pendingKey, pendingByteArray),
+                                put(stablePartAssignmentsKey(partId), assignmentsByteArray),
+                                put(changeTriggerKey, rev)
+                        ).yield(),
+                        If.iif(
+                            and(
+                                    or(notExists(changeTriggerKey), value(changeTriggerKey).lt(rev)),
+                                    notExists(pendingKey)
+                            ),
+                            Operations.ops(
+                                    put(pendingKey, pendingByteArray),
+                                    put(changeTriggerKey, rev)
+                            ).yield(),
+                            ops().yield()
```

Comment:
Hm, what does this do?

Reply:
Just nothing

Reply:
Oh, it's an `else` branch! That is SO hard to read.
You know what? We can do what Mockito does, in a way. Add methods like `then` and `els` just for sugar.

Reply:
I think it's a good idea, yes